chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -63,6 +63,12 @@ if [ $(id -u) -gt 0 ]; then
     exit 1
 fi
 
+if test -n "$SSH_CONNECTION" -a -z "$STY"; then
```

Comment:
test and [ is identical with any current shell performance wise. as you noticed, there is no predominant style yet so it doesn't make sense to refuse the PR on that base yet.